src/terrain/generation/map_tile.hpp

```diff
@@ -0,0 +1,49 @@
+#include "../../types.hpp"
+
+#include <mutex>
+#include <random>
+
+#pragma once
+
+namespace terrain {
+
+namespace generation {
+
+class MapTile {
+ private:
+    MacroDim x_; // x position
+    MacroDim y_; // y position
+
+    MapTile_t tile_type_; // map tile type
+    // TODO add mutex
+    // This requires >= 64 x 64 map macro tiles. This would correspond to
```

Comment:
wat?

Reply:
The vector must be larger than 64x64 for multiprocessing to be beneficial. Therefore we should not add a mutex. For optimization it would be better to parallelize the meaher (no data race).

Reply:
no i meant the second half of this comment is missing